### PR TITLE
Add first Korean cookbook

### PR DESCRIPTION
**Pull request by harheem**
# What does this PR do?
This PR represents the first step in adding Korean content to the Open-Source AI Cookbook. The main changes are as follows:
1. Created 'ko' directory: A new 'ko' directory has been added under the notebooks folder to house Korean content.
2. Added Korean Advanced RAG notebook: This notebook introduces advanced RAG (Retrieval-Augmented Generation) techniques to Korean-speaking users.
3. Created table of contents and index files:
   - Generated the notebooks/ko/_toctree.yml file.
   - Added the notebooks/ko/index.md file.
4. Updated workflow files: Added the 'ko' language code to build_documentation.yml and build_pr_documentation.yml files in the .github/workflows folder.

## Who can review?
@jungnerd, @heuristicwave, @junejae, @nuatmochoi
@merveenoyan, @stevhliu.

### CODE REVIEW

**Review by merveenoyan — COMMENTED** [review]
Very comprehensive, gomapseumnida! 🤗
Would be great for a Korean speaking reviewer to skim through it. Left two small nits.

**Comment by harheem** [comment]
Thank you all for your reviews! @merveenoyan @ jungnerd ❤️
I've incorporated the feedback you provided through ReviewNB.

File: notebooks/ko/_toctree.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  isExpanded: True
+  sections:
+    - local: index
+      title: 오픈소스 AI Cookbook
+      
+    - title: LLM 레시피
+      isExpanded: false
+      sections:
+        - local: advanced_ko_rag
+          title: 한국어로 Advanced RAG 구현하기 - Hugging Face와 LangChain 활용한 Cookbook
```
Comment (heuristicwave) [inline notebooks/ko/_toctree.yml:11]:
```suggestion
          title: 한국어로 Advanced RAG 구현하기 - Hugging Face와 LangChain을 활용한 Cookbook
```

Comment (heuristicwave) [reply notebooks/ko/_toctree.yml:11]: ipynb 파일 제목에서도 '을' 조사를 추가하는 것과, 맨 마지막 Kiwi 부분에 대한 설명만 추가되면 좋을 것 같습니다 :)

**Comment by harheem** [comment]
@merveenoyan, @stevhliu
Would you mind considering merging this PR? 😊

**Review by stevhliu — APPROVED** [review]
LGTM! Thanks for kicking off Korean contributions for the cookbook! 🤗

**Pull request merged**